### ISSUE TITLE
zOù MAï sequences always snap to multiples of 4

### ISSUE THREAD

**Issue by MaZe6K** [sebastien-bouffier/Bidoo#148]
OS (if applicable): Windows 10
Version (or "dev" if compiling from source): 1.1.15

Hi there!
I'm playing with zOù MAï again and just realised that it doesn't react well if I try to have sequences with irregular lengths.
For example if I want to have one sequence with 3 steps length, it plays 1-2-3-1, and then repeats with 1-2-3-1-1-2-3-1-1-2-3-1-.....
If I set the sequence length to 9 it plays 2-3-4-5-6-7-8-9-2-3-4-5-6-7-8-9-... and so on, it somehow always snaps to a value based on 4.
I think it reacted different when it still had the syncing problems, so I guess it has something to do with your solution for that?

**Comment by MaZe6K** [comment]
addendum: Individual channel reset also only reacts in steps of 4, and on those snapped sequences the gate is usually missing on the first step.

**Comment by sebastien-bouffier** [comment]
Hi @MaZe6K ,
Regarding your first issue i.e. 1-2-3-1-1-2-3-1-1 it seems that you reset each four steps so you obtain that kind of pattern. When I set a loop of 3 steps it loops on 1-2-3-1-2-3-1-ect until I reset.
I can not reproduce the second issue with the global reset ... but .... but ...but if I use the track reset I see that the first step is omitted. I will fix that issue shortly.
Thanks for your interest in my modules :)
If you can please send me your patch with the issue you describe, which module do you use as clock ?
Best.

**Comment by MaZe6K** [comment]
It has something to do with the speed of the channel. At 1x speed it seems to work fine.
I had it at 4x speed, where it plays the 3 steps as I explained, but I think it doesn't play the first step twice, but twice as long, because there is only one gate on step 1.
At 2x speed it only plays steps 2-3-2-3... At 3x speed it's fine again... With higher speeds it gets more extreme and rests on step 1 for a longer time.
I would upload an example, but I don't think I can upload it here, I just made a test patch again with only Impromptu CLKD as clock and zOù MAï driving a very simple voice, no reset used, so I think it always does this.
If you experiment with different speeds and steplengths it plays different variations of the pattern.
